### PR TITLE
Add saving support to Settings file

### PR DESCRIPTION
**Pull request by zorg93**
Also Add method called after constructor has finished that can be overridden called OnLoaded, This means you can access ScriptSettings before the main script starts without getting errors that you would get by putting it in the constructor
fixes these issues
https://github.com/crosire/scripthookvdotnet/issues/249
https://github.com/crosire/scripthookvdotnet/issues/248

### CODE REVIEW

File: source/Settings.cpp [hunk 1]
```diff
@@ -198,5 +197,60 @@ namespace GTA
...
 		this->mValues->default[lookup] = value;
+		if (!this->mValues->ContainsKey(lookup))
+			this->mValues->Add(lookup, value);
+		else
+			this->mValues->default[lookup] = value;
+	}
+	void ScriptSettings::Save()
+	{
+		if (mFileName == nullptr)
+			throw gcnew Exception("The file name is still fucking null when saving");
```
Comment (IncoCode) [inline source/Settings.cpp]: :confused:

Comment (zorg93) [inline source/Settings.cpp]: I got frustrated, I should remove that

**Comment by JohnnyCrazy** [comment]
Looks good to me @crosire 

Devs can decide if they want to support saving (with the disadvantage of comments getting deleted) or don't want to support it (and keep the comments)

**Comment by ardaozkal** [comment]
I support it. It is at least better than this:    

File.WriteAllText(configfile, ("[" + sectionname + "]"+ Environment.NewLine + "Enable/Disable_Menu = F6" + Environment.NewLine + "Forward_on_noclip = W" + Environment.NewLine + "Back_on_noclip = S" + Environment.NewLine + "Left_on_noclip = A" + Environment.NewLine + "Right_on_noclip = D" + Environment.NewLine + "Increase_speed_on_noclip = Add" + Environment.NewLine + "Increase_speed_on_noclip = Subtract"));

**Pull request merged**